Lib/shutil.py

```diff
@@ -136,6 +136,7 @@ def _fastcopy_sendfile(fsrc, fdst):
     # changes while being copied.
     try:
         blocksize = max(os.fstat(infd).st_size, 2 ** 23)  # min 8MB
+        blocksize = min(blocksize, 2 ** 31 - 2)  # max 2GB
```

Comment:
Why not `2 ** 31 - 1`?

Would not be better to use a power of two, `2**30`?

Reply:
Agreed, used `2**30`.

Reply:
That sounds inefficient on 64-bit system: files larger than 2 GB will need multiple sendfile() syscalls, rather currently a single one is enough, no? Maybe use sys.maxsize instead?

Reply:
On my 64-bit Ubuntu 18.04 box sendfile() always copies up to 2GB per call, never more than that.

Reply:
OK, I think it makes sense. I added:

```diff
+        if sys.maxsize < 2 ** 32:  # 32-bit architecture
+            blocksize = min(blocksize, 2 ** 30)  # max 1GiB
```

Reply:
...I did that for shutil only. In `socket.py` I decided to always limit to max 1GiB: since we're dealing with a socket I want to exercise the loop more often (and hit the selector(), which implements the timeout logic).